knowledge_graphs/ai_script_analyzer.py

```
@@ -0,0 +1,550 @@
+        suspicious_names = [
+            'hack', 'crack', 'exploit', 'bypass', 'steal', 'leak',
+            'magic_solve', 'auto_fix_everything', 'instant_solution'
+        ]
+        
+        for func in result["functions"]:
+            if any(sus in func["name"].lower() for sus in suspicious_names):
+                suspicious.append(f"suspicious_function_name:{func['name']}")
+        
+        result["suspicious_patterns"] = suspicious
+    
+    def _get_decorator_name(self, decorator):
```

Ruff (0.12.2)

Missing return type annotation for private function `_get_decorator_name`

(ANN202)